crates/miden-objects/src/transaction/proven_tx.rs

```diff
@@ -133,6 +136,85 @@ impl ProvenTransaction {
+    /// - The size of the serialized account update exceeds [`ACCOUNT_UPDATE_MAX_SIZE`].
+    /// - The transaction is empty, which is the case if the account state is unchanged or the
+    ///   number of input notes is zero.
+    /// - The transaction was executed against a _new_ account with public state and its account ID
+    ///   does not match the ID in the account update.
+    /// - The transaction was executed against a _new_ account with public state and its commitment
+    ///   does not match the final state commitment of the account update.
+    /// - The transaction was executed against a private account and the account update is _not_ of
+    ///   type [`AccountUpdateDetails::Private`].
+    /// - The transaction was executed against an account with public state and the update is of
+    ///   type [`AccountUpdateDetails::Private`].
+    /// - The transaction was executed against an _existing_ account with public state and the
+    ///   update is of type [`AccountUpdateDetails::New`].
+    /// - The transaction creates a _new_ account with public state and the update is of type
+    ///   [`AccountUpdateDetails::Delta`].
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        account_id: AccountId,
+        initial_account_commitment: Word,
+        final_account_commitment: Word,
+        account_delta_commitment: Word,
+        account_update_details: AccountUpdateDetails,
+        input_notes: impl IntoIterator<Item = InputNoteCommitment>,
+        output_notes: impl IntoIterator<Item = OutputNote>,
+        ref_block_num: BlockNumber,
+        ref_block_commitment: Word,
+        fee: FungibleAsset,
+        expiration_block_num: BlockNumber,
+        proof: ExecutionProof,
+    ) -> Result<Self, ProvenTransactionError> {
+        let input_notes_vec: Vec<_> = input_notes.into_iter().collect();
+        let output_notes_vec: Vec<_> = output_notes.into_iter().collect();
+
+        let input_notes =
+            InputNotes::new(input_notes_vec).map_err(ProvenTransactionError::InputNotesError)?;
+        let output_notes =
+            OutputNotes::new(output_notes_vec).map_err(ProvenTransactionError::OutputNotesError)?;
+        let id = TransactionId::new(
+            initial_account_commitment,
+            final_account_commitment,
+            input_notes.commitment(),
+            output_notes.commitment(),
+        );
+        let account_update = TxAccountUpdate::new(
+            account_id,
+            initial_account_commitment,
+            final_account_commitment,
+            account_delta_commitment,
+            account_update_details,
+        );
+
+        let proven_transaction = Self {
+            id,
+            account_update,
+            input_notes,
+            output_notes,
+            ref_block_num,
+            ref_block_commitment,
+            fee,
+            expiration_block_num,
+            proof,
+        };
+
+        proven_transaction.validate()
```

Comment:
I would move the validation code of `validate` here and remove the `validate` method completely. But, because this `validate` has changed significantly in #1896, I would not do this before the PR is merged or until you've rebased on top of that PR.